src/library/scala/util/Random.scala

```diff
@@ -70,6 +70,16 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    */
   def nextLong(): Long = self.nextLong()
 
+  def nextLong(n: Long): Long = {
+    if (n <= 0) throw new java.lang.IllegalArgumentException("bound must be positive")
+    if (n <= Int.MaxValue) nextInt(n.toInt).toLong
+    else {
+      val RAND_MAX = Long.MaxValue     // 2^63-1
+      val x = nextLong() >>> 1         // [0, 2^63), strip the sign bit
+      if (x < (RAND_MAX - (RAND_MAX % n))) x % n else nextLong(n)
```

Comment:
Yep, that's the best of Scala.js ;)

Benchmarking in Scala.js is not trivial. You can have a look at https://github.com/sjrd/scalajs-benchmarks for inspiration. The easiest for quick tests is to modify an existing benchmark (like Richards, which is pretty self-contained) and replace it with what you want to test instead. Then run `richardsJS/run`.